src/ray/util/filesystem.h

```diff
@@ -22,68 +23,24 @@
 
 namespace ray {
 
-/// \return The portable directory separator (slash on all OSes).
-static inline char GetAltDirSep() { return '/'; }
-
 /// \return The platform directory separator (backslash on Windows, slash on other OSes).
-static inline char GetDirSep() {
-  char result;
-#ifdef _WIN32
-  result = '\\';
-#else
-  result = '/';
-#endif
-  return result;
-}
-
-/// \return The platform PATH separator (semicolon on Windows, colon on other OSes).
-static inline char GetPathSep() {
-  char result;
-#ifdef _WIN32
-  result = ';';
-#else
-  result = ':';
-#endif
-  return result;
-}
-
-/// Returns the executable binary suffix for the platform, if any.
-std::string GetExeSuffix();
+static inline char GetDirSep() { return std::filesystem::path::preferred_separator; }
 
 /// Equivalent to Python's os.path.basename() for file system paths.
 std::string GetFileName(const std::string &path);
 
-size_t GetRootPathLength(const std::string &path);
-
-/// \return A non-volatile temporary directory in which Ray can stores its files.
-std::string GetRayTempDir();
-
 /// \return The non-volatile temporary directory for the current user (often /tmp).
 std::string GetUserTempDir();
 
 /// \return Whether or not the given character is a directory separator on this platform.
 static inline bool IsDirSep(char ch) {
-  bool result = ch == GetDirSep();
-#ifdef _WIN32
-  result |= ch == GetAltDirSep();
-#endif
-  return result;
+  return ch == std::filesystem::path::preferred_separator;
```

Comment:
This is not equivalent to the original. On Win32 `/` is a valid separator as well.

Reply:
recovered to the original logic.